play.pokemonshowdown.com/src/panel-teamdropdown.tsx

```diff
@@ -532,6 +535,24 @@ export class PSTeambuilder {
 
 		return team;
 	}
+
+	static formatResources = {} as Record<string, FormatResource>;
+
+	static getFormatResources(format: string): Promise<FormatResource> {
+		if (format in this.formatResources) return Promise.resolve(this.formatResources[format]);
+		return new Promise(resolve => {
+			Net('https://www.smogon.com/dex/api/formats/by-ps-name/' + format).get().then(
+				result => {
+					this.formatResources[format] = JSON.parse(result);
+					resolve(this.formatResources[format]);
+				},
+				err => {
+					this.formatResources[format] = null;
+					resolve(this.formatResources[format]);
+				}
+			);
+		});
```

Comment:
I guess you're not used to pre-async/await Promises. Valid, they both landed roughly the same time.

```suggestion
		if (format in this.formatResources) return Promise.resolve(this.formatResources[format]);
		return Net('https://www.smogon.com/dex/api/formats/by-ps-name/' + format).get()
			.then(result => {
				this.formatResources[format] = JSON.parse(result);
				return this.formatResources[format];
			}).catch(err => {
				this.formatResources[format] = null;
				return this.formatResources[format];
			});
```

Reply:
Nope, that was literally just me writing `new Promise`, remembering `Net` is already promise, and then forgetting to remove it.